actionpack/test/controller/action_pack_assertions_test.rb

```diff
@@ -170,6 +170,16 @@ def test_string_constraint
     end
   end
 
+  def test_with_routing_works_with_api_only_controllers
+    @controller = Class.new(ActionController::API).new
+
+    with_routing do |set|
+      set.draw do
+        get "wow", to: "action_pack_assertions#nothing"
+      end
```

Comment:
Hi! I've changed the test so it does the same as the one you put as example :) 
I didn't do the assertion before because the test was failing by just using `with_routing`